cmp/compare.go

```diff
@@ -377,15 +377,18 @@ func (s *state) compareArray(vx, vy reflect.Value, t reflect.Type) {
 	s.curPath.push(step)
 
 	// Compute an edit-script for slices vx and vy.
-	eq, es := diff.Difference(vx.Len(), vy.Len(), func(ix, iy int) diff.Result {
+	es := diff.Difference(vx.Len(), vy.Len(), func(ix, iy int) diff.Result {
 		step.xkey, step.ykey = ix, iy
 		return s.statelessCompare(vx.Index(ix), vy.Index(iy))
 	})
 
-	// Equal or no edit-script, so report entire slices as is.
-	if eq || es == nil {
+	// Report the entire slice as is if the arrays are of primitive kind,
+	// and the arrays are different enough.
+	k := t.Elem().Kind()
+	isPrimitive := reflect.Bool <= k && k <= reflect.Complex128
```

Comment:
Relying on the order of constants in the reflect package doesn't seem like a great idea.

Reply:
Done